whitelist.py

```diff
@@ -0,0 +1,39 @@
+# Because vulture only performs static analysis, this file is used
+# by vulture as a whitelist module and passed as a parameter when
+# running vulture so that it'll know that certain methods are being
+# used.
+
+from density import density
+from bokeh import figure
+
+# Ignore bokeh figure attribute assignment
+figure.xaxis.axis_label
+figure.xaxis.axis_line_width
+figure.xaxis.axis_line_color
+figure.xaxis.major_label_text_color
+figure.yaxis.axis_label
+figure.yaxis.axis_line_color
+figure.yaxis.major_label_text_color
+figure.yaxis.major_label_orientation
+figure.yaxis.axis_line_width
+
+# Ignore Flask app attribute assignment
+density.app.json_encoder
+
+# Ignore all Flask routes
+density.get_connections
```

Comment:
I've been experimenting, and definitely not; the static analysis performed by vulture imho isn't really extensive at all. It ignores scopes and only performs on object names; in particular, it performs analysis by building a syntax tree, but the problem is that you can spoof the names using random objects.

For example, `density.map` is unused according to vulture, but you can suppress this warning by simply declaring it in a whitelist module. But the `density` portion doesn't even need to exist; in fact, turns out it doesn't even need to be imported. Replacing the `density.map` declaration in the whitelist with something like `banana.map` similarly suppresses the warning.

To be honest, at this point I would actually be fine with not including vulture in the testing process; it seems like it's more trouble than it's worth.